### PR TITLE
Change `AutoMLSearch` parameters for AutoML Algorithm Upgrade

### PR DESCRIPTION
**Pull request by jeremyliweishih**
Adds on to #3261.

Changes:

- Set default max batches in search to 4 batches (3 for time series), which corresponds to fast mode
- Make `_automl_algorithm` parameter public

### CODE REVIEW

File: evalml/automl/automl_search.py [hunk 1]
```diff
@@ -559,7 +559,7 @@ def __init__(
         self.max_batches = max_batches
         self._pipelines_per_batch = _pipelines_per_batch
         if not self.max_iterations and not self.max_time and not self.max_batches:
-            self.max_batches = 1
+            self.max_batches = 4 if automl_algorithm == "default" else 1
```
Comment (dsherry) [inline evalml/automl/automl_search.py:562]: @jeremyliweishih could this be an attribute of the automl algo?
```python
self.max_batches = automl_algorithm.default_max_batches
```

File: evalml/tests/automl_tests/test_automl_iterative_algorithm.py [hunk 2]
```diff
@@ -425,8 +426,13 @@ def test_max_batches_num_pipelines(
             + num_ensemble_batches
         )
         n_automl_pipelines = n_results
-    assert automl._automl_algorithm.batch_number == max_batches
-    assert automl._automl_algorithm.pipeline_number + 1 == n_automl_pipelines
+    if max_batches is None:
```
Comment (jeremyliweishih) [inline evalml/tests/automl_tests/test_automl_iterative_algorithm.py:429]: This is a weird one and let me try to explain! In this PR, I make[ these changes ](https://github.com/alteryx/evalml/pull/3304/files#diff-8b4df814812e6f5df24c600bc7f429b036e0e2f78613bfbdcdf0ed3ab86bffd6R732-R736) which not only set the default behavior for max batches but more importantly moved this logic after the automl algorithm is instantiated. So, on main, the default max batches is set before the automl algorithm is set. This meant that if `max_batches`, `max_iterations` etc. were all `None` on main, `max_batches` would be set to 1 and would be passed into `IterativeAlgorithm`. However, `IterativeAlgorithm` has [this logic ](https://github.com/alteryx/evalml/blob/main/evalml/automl/automl_algorithm/iterative_algorithm.py#L227-L254) which sets `max_iterations` if `ensembling = True` and if `max_batches is not None`. This meant that on `main`  , turning `ensembling=True` would mean that `AutoMLSearch` would use `max_iterations` as stopping criteria. When I made the changes to search in this PR, it meant that search was correctly using `max_batches` as stopping criteria which meant that search would stop when `batch_number > max_batches`. These changes to this test reflect that and with an added complication of `_pipelines_per_batch`.

Comment (jeremyliweishih) [reply evalml/tests/automl_tests/test_automl_iterative_algorithm.py:429]: Think the above makes sense but still have some details to iron out.

File: evalml/automl/automl_algorithm/default_algorithm.py [hunk 3]
```diff
@@ -145,6 +144,11 @@ def __init__(
                         " and Real!"
                     )
 
+    @property
+    def default_max_batches(self):
+        """Returns the number of max batches AutoMLSearch should run by default."""
+        return 4 if not is_time_series(self.problem_type) else 3
```
Comment (jeremyliweishih) [inline evalml/automl/automl_algorithm/default_algorithm.py:150]: Will need to update after ##3191.

**Review by chukarsten — APPROVED** [review]
This LGTM, Jeremy!  I think as a suggestion to anyone reading this comment, we should try to keep small and less significant changes (like publicizing automl_algorithm) in separate PRs to maximize views on perhaps the more impactful parts of a PR.  Thanks for doing this, though!  Just trying to increase visibility on the more important stuff rather than the tedious!

File: evalml/tests/automl_tests/test_automl.py [hunk 4]
```diff
@@ -2600,7 +2598,7 @@ def test_search_with_text(AutoMLTestEnv):
     env = AutoMLTestEnv("binary")
     with env.test_context(score_return_value={"Log Loss Binary": 0.30}):
         automl.search()
-    assert automl.rankings["pipeline_name"][1:].str.contains("Natural Language").all()
+    assert automl.rankings["pipeline_name"][1:-1].str.contains("Natural Language").all()
```
[the remark this answers is not in the document]
Comment (jeremyliweishih) [reply evalml/tests/automl_tests/test_automl.py:2601]: Its the ensembling pipeline so it won't contain `Natural Language` in the pipeline name.

**Pull request merged**